src/min-reviewers.ts

```diff
@@ -0,0 +1,101 @@
+import { Label, PullContext, Review } from "./types";
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+
+export function getMinReviewers(labels: Label[]): string | number {
+  const pattern = /min-(?<number>\d|all)-reviewers/;
+
+  for (const label of labels) {
+    const m = label.name.match(pattern);
+    if (!m) {
+      continue;
+    }
+
+    const { groups = {} } = m;
+    const { number = "0" } = groups;
+    return number === "all" ? number : parseInt(number, 10);
+  }
+
+  return 0;
+}
+
+export function requirementPassed(
+  reviews: Review[],
+  requestedReviewers: number,
+  minReviewers: number | string
+): boolean {
+  const latestReviews = reviews
+    .reverse()
+    .filter((review) => review.state.toLowerCase() !== "commented")
+    .filter((review, index, array) => {
+      // https://dev.to/kannndev/filter-an-array-for-unique-values-in-javascript-1ion
+      return array.findIndex((x) => review.user?.id === x.user?.id) === index;
+    });
+
+  const approvedReviews = latestReviews.filter(
+    (review) => review.state.toLowerCase() === "approved"
+  );
+  core.info(`Approved reviews: ${approvedReviews.length}. Required: ${minReviewers}`);
+
+  if (minReviewers === "all") {
+    core.debug(`Pull request reviewers: ${requestedReviewers}`);
+
+    // some reviewers have not reviewed
+    if (requestedReviewers > 0) {
+      return false;
+    }
+
+    // some reviewers do not approve
+    if (!latestReviews.every((review) => review.state.toLowerCase() === "approved")) {
+      return false;
+    }
+  }
+
+  return approvedReviews.length >= minReviewers;
+}
+
+export async function run(): Promise<void> {
+  const token: string = core.getInput("GITHUB_TOKEN", { required: true });
+
+  if (token === "") {
```

Comment:
This could be just `!token`, because in javascript empty string is considered as `null`. This is not required change, but just in case somehow token becomes `undefined` | `null` we would have a check for that as well.